code/game/objects/items/weapons/airlock_painter.dm

```diff
@@ -33,14 +33,62 @@
 //because you're expecting user input.
 /obj/item/weapon/airlock_painter/proc/can_use(mob/user)
 	if(!ink)
-		user << "<span class='notice'>There is no toner cardridge installed installed in \the [name]!</span>"
+		user << "<span class='notice'>There is no toner cartridge installed installed in \the [name]!</span>"
 		return 0
 	else if(ink.charges < 1)
 		user << "<span class='notice'>\The [name] is out of ink!</span>"
 		return 0
 	else
 		return 1
 
+/obj/item/weapon/airlock_painter/suicide_act(mob/user)
+	var/has_lungs = FALSE
+	if(user.getorganslot("lungs"))
+		has_lungs = TRUE
+
+	if(can_use(user) && has_lungs)
+		user.visible_message("<span class='suicide'>[user] is inhaling toner from \the [name]! It looks like \he's trying to commit suicide.</span>")
+		use(user)
+
+		// Once you've inhaled the toner, you throw up your lungs
+		// and then die.
+
+		// Find out if there is an open turf in front of us,
+		// and if not, pick the turf we are standing on.
+		var/turf
+
+		turf = get_step(get_turf(src), user.dir)
+		if(!istype(turf, /turf/open))
+			turf = get_turf(src)
+
+		var/obj/item/organ/internal/lungs/L
+		L = user.getorganslot("lungs")
+		L.Remove(user)

```

Comment:
Make sure L isn't null, in the event someone already lost their lungs, or this could runtime.